gremlin-console/src/main/groovy/org/apache/tinkerpop/gremlin/console/Console.groovy

```diff
@@ -191,7 +192,18 @@ class Console {
             groovy.setResultHook(handleResultShowNothing)
     }
 
-    private def handlePrompt = { interactive ? STANDARD_INPUT_PROMPT : "" }
+    private def handlePrompt = { 
+        if (interactive) {
+            int lineNo = groovy.buffers.current().size()
+            if (lineNo > 0 ) {
+                return lineNo.toString().padLeft(STANDARD_INPUT_PROMPT_PAD, '0').padRight(STANDARD_INPUT_PROMPT.length()-2, ' ') + "> "
```

Comment:
if it's easier to evaluate as merged, then i think we might want to complete the vote here for this work, as we already have three votes on the other PR.